app/controllers/api/v1/applications/sources/scenarios_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Api
+  module V1
+    module Applications
+      module Sources
+        class ScenariosController < Api::V1::Applications::Sources::ApplicationController
+          def index
+            scenario = event_source.scenarios.find(params[:id])
+
+            rows = Event.find_by(
+              application: application,
+              event_source: event_source,
+              scenario: scenario
+            )
+
+            response.headers["Access-Control-Allow-Origin"] = "*"
+
+            events = rows.map do |row|
```

Comment:
Unused block argument - `row`. You can omit the argument if you don't care about it.<br>Useless assignment to variable - `events`. Did you mean `event_source`?